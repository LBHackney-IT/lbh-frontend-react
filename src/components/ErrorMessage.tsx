import classNames from "classnames";
import PropTypes from "prop-types";
import React, { FunctionComponent, ReactNode } from "react";

import "lbh-frontend/lbh/components/lbh-error-message/_error-message.scss";

export type ErrorMessageProps = {
  id?: string;
  className?: string;
  visuallyHiddenText?: string;
  children?: ReactNode;
};

export const ErrorMessage: FunctionComponent<ErrorMessageProps> = ({
  id,
  className,
  visuallyHiddenText,
  children
}) => (
  <span
    id={id}
    className={classNames(
      "govuk-error-message",
      "lbh-error-message",
      className
    )}
  >
    {visuallyHiddenText && (
      <span className="govuk-visually-hidden">{visuallyHiddenText}:</span>
    )}
    {children}
  </span>
);

ErrorMessage.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  visuallyHiddenText: PropTypes.string,
  children: PropTypes.node
};

ErrorMessage.defaultProps = {
  visuallyHiddenText: "Error"
};
