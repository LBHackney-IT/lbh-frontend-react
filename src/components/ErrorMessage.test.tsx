import React from "react";
import { create } from "react-test-renderer";

import { ErrorMessage } from "./ErrorMessage";

it("renders correctly with all props", () => {
  const component = create(
    <ErrorMessage
      id="1234"
      className="class1 class2"
      visuallyHiddenText="visually hidden text"
    >
      test <strong>error</strong> message
    </ErrorMessage>
  );

  expect(component).toMatchInlineSnapshot(`
    <span
      className="govuk-error-message lbh-error-message class1 class2"
      id="1234"
    >
      <span
        className="govuk-visually-hidden"
      >
        visually hidden text
        :
      </span>
      test 
      <strong>
        error
      </strong>
       message
    </span>
  `);
});

it("renders correctly without optional props", () => {
  const component = create(<ErrorMessage />);

  expect(component).toMatchInlineSnapshot(`
    <span
      className="govuk-error-message lbh-error-message"
    >
      <span
        className="govuk-visually-hidden"
      >
        Error
        :
      </span>
    </span>
  `);
});
