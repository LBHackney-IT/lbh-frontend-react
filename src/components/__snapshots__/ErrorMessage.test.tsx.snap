// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with all props 1`] = `
<span
  className="govuk-error-message lbh-error-message class1 class2"
  id="1234"
>
  <span
    className="govuk-visually-hidden"
  >
    visually hidden text
    :
  </span>
  test 
  <strong>
    error
  </strong>
   message
</span>
`;

exports[`renders correctly without optional props 1`] = `
<span
  className="govuk-error-message lbh-error-message"
>
  <span
    className="govuk-visually-hidden"
  >
    Error
    :
  </span>
</span>
`;
